src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLanguageServer.cs

```diff
@@ -59,30 +58,40 @@ private RazorLanguageServer(ILanguageServer innerServer)
 
         public static Task<RazorLanguageServer> CreateAsync(Stream input, Stream output, Trace trace, Action<RazorLanguageServerBuilder> configure = null)
         {
-            Serializer.Instance.Settings.Converters.Add(SemanticTokensOrSemanticTokensEditsConverter.Instance);
+            // Serializer.Instance.Settings.Converters.Add(SemanticTokensOrSemanticTokensEditsConverter.Instance);
             Serializer.Instance.JsonSerializer.Converters.RegisterRazorConverters();
 
             // Custom ClientCapabilities deserializer to extract experimental capabilities
             Serializer.Instance.JsonSerializer.Converters.Add(ExtendableClientCapabilitiesJsonConverter.Instance);
 
             ILanguageServer server = null;
+            var logLevel = trace switch
+            {
+                Trace.Messages => LogLevel.Information,
+                Trace.Off => LogLevel.None,
+                Trace.Verbose => LogLevel.Trace,
+                _ => default,
+            };
+
             server = OmniSharp.Extensions.LanguageServer.Server.LanguageServer.PreInit(options =>
                 options
                     .WithInput(input)
                     .WithOutput(output)
+                    .WithContentModifiedSupport(false)
```

Comment:
What's this and why's it needed?

Reply:
*Quietly weeping* OK, so StreamJsonRpc has both Serial and Parallel requests. With `WithContentModifiedSupport(true)` (which is default) when a Serial request is made any Parallel requests will be cancelled because the assumption is that Serial requests modify state, and that therefore any Parallel request is now invalid and should just try again. A specific instance of this can be seen when you hover over a TagHelper while the switch is set to true. Hover is parallel, and a lot of our endpoints like `textDocument/_ms_onAutoInsert`, and `razor/languageQuery` are Serial. I BELIEVE that specifically what happened is the serial languageQuery event gets fired by our semantic tokens endpoint (which fires constantly), cancelling the hover, which red-bars. We can prevent that behavior entirely by doing `WithContentModifiedSupport`, at the possible expense of some delays due doing all requests in serial.

I DO recommend that we attempt to resolve this and switch back to `WithContentModifiedSupport(true)` in the future, but I think it would either mean having 0 Serial Handlers in the whole LS, or making VSLanguageServerClient handle this more gracefully, and I didn't figure we had time to deal with either of those for this PR.

Many Bothans died to bring us this information.

Reply:
Love the explanation! Please turn this into a comment so future readers understand